### PR TITLE
Use minimum access profile in UTIL_Profile when available

### PR DESCRIPTION
**Pull request by psadie-sfdo**
We made changes to prevent unit test failures once the 'Read Only' profile is removed from our persistent orgs.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/UTIL_Profile.cls [hunk 1]
```diff
@@ -83,27 +83,57 @@ public class UTIL_Profile {
...
+            if (hasReadOnlyProfile == true) {
+                return readOnly;
+            } else {
+                return PROFILE_MINIMUM_ACCESS;
+            }
+        }
+    }
+
+    /** @description True if the Read Only profile exists in the org */
+    private static Boolean hasReadOnlyProfile;
```
Comment (bdvorachek) [inline src/classes/UTIL_Profile.cls:125]: I think this definition fits better up above the PROFILE_READ_ONLY getter.

File: src/classes/UTIL_Permissions_TEST.cls [hunk 2]
```diff
@@ -47,6 +47,177 @@ public with sharing class UTIL_Permissions_TEST {
...
+    @isTest
+    static void shouldReturnFalseForUserWithoutDeleteAccessToCustomFieldByString() {
+        System.runAs(getReadOnlyUser()) {
+            System.assert(!UTIL_Permissions.canUpdate(customObjectName, customFieldName, false),
+                'Minimum access user should not have delete access to custom field.');
+        }
+    }
+
+    @isTest
+    static void shouldReturnTrueForUserWithoutCreateAccessAfterPermissionAssignment() {
```
Comment (bdvorachek) [inline src/classes/UTIL_Permissions_TEST.cls:192]: I feel like we're testing platform functionality with this one and can pull it. I checked and it doesn't seem to cover any lines that are not already covered by another test.

**Review by bdvorachek — CHANGES_REQUESTED** [review]

File: src/classes/UTIL_Profile.cls (hunk 1 above)
Comment (psadie-sfdo) [reply src/classes/UTIL_Profile.cls]: Updated!

File: src/classes/UTIL_Permissions_TEST.cls (hunk 2 above)
Comment (psadie-sfdo) [reply src/classes/UTIL_Permissions_TEST.cls]: Updated!

**Comment by psadie-sfdo** [comment]
GUS Work Item: [W-8543149](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008kv47IAA/view)

**Review by bdvorachek — APPROVED** [review]
Coverage looks good, tests passing in both a prerelease and regular scratch org.

File: src/classes/UTIL_Permissions_TEST.cls [hunk 3]
```diff
@@ -47,6 +47,147 @@ public with sharing class UTIL_Permissions_TEST {
...
+        }
+    }
+
+    @isTest
+    static void shouldThrowExceptionForUserWithoutEditAccessToObjectByString() {
+        System.runAs(getReadOnlyUser()) {
+            try {
+                UTIL_Permissions.canUpdate(standardObjectName);
+            } catch (Exception ex) {
+                System.assert(ex instanceof UTIL_Permissions.InsufficientPermissionException);
```
Comment (vmartinov-sforg) [inline src/classes/UTIL_Permissions_TEST.cls:105]: Let's add a message to the assert, so it's consistent with the other methods.

**Review by vmartinov-sforg — COMMENTED** [review]
Looks good - I'm running all the tests now. I only have a few minor comments.

File: src/classes/UTIL_Permissions_TEST.cls (hunk 3 above)
Comment (psadie-sfdo) [reply src/classes/UTIL_Permissions_TEST.cls]: updated!

File: src/classes/UTIL_Profile.cls [hunk 4]
```diff
@@ -83,27 +83,57 @@ public class UTIL_Profile {
...
             }
-            return 'Read Only';
+
+            if (hasReadOnlyProfile == null) {
+                hasReadOnlyProfile = hasProfile(readOnly);
+            }
+
+            if (hasReadOnlyProfile == true) {
+                return readOnly;
+            } else {
```
[the remark this answers is not in the document]
Comment (psadie-sfdo) [reply src/classes/UTIL_Profile.cls:121]: Updated!

File: src/classes/UTIL_Profile.cls [hunk 5]
```diff
@@ -83,28 +83,57 @@ public class UTIL_Profile {
...
+                    readOnly = 'Alleen-lezen';
                 }
             }
-            return 'Read Only';
+
+            if (hasReadOnlyProfile == null) {
+                hasReadOnlyProfile = hasProfile(readOnly);
+            }
+
+            if (hasReadOnlyProfile == true) {
```
Comment (vmartinov-sforg) [inline src/classes/UTIL_Profile.cls:119]:
```suggestion
            if (hasReadOnlyProfile == true) {
                return readOnly;
            } 
            return PROFILE_MINIMUM_ACCESS;
```

**Review by vmartinov-sforg — COMMENTED** [review]
Looks good - one last suggestion.

File: src/classes/UTIL_Profile.cls (hunk 5 above)
Comment (psadie-sfdo) [reply src/classes/UTIL_Profile.cls:122]: Updated!

**Review by vmartinov-sforg — APPROVED** [review]
Looks good, thank you Prince!

**Review by ptewson-sfdo — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**